### PR TITLE
[rlp] Add no_std support 

### LINKED ISSUE #171: [rlp] convert benches to criterion

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

# What have I changed?

- Add `no_std` support for `rlp` crate
- Fix some warnings
- Convert `rlp` benches to `criterion`

related issue: #167, closes #171

### CODE REVIEW

**Comment by koushiro** [comment]
Sorry to reply so late. I will split #167 into multiple PRs. PTAL @dvdplm

**Comment by koushiro** [comment]
Does anyone can review my code?

File: .travis.yml [hunk 1]
```diff
@@ -28,13 +28,13 @@ script:
...
-  - cd fixed-hash/ && cargo test --all-features && cd ..
-  - cd uint/ && cargo test --features=std,quickcheck --release && cd ..
+  - cd fixed-hash/ && cargo test --all-features && cargo test --no-default-features --features="byteorder,rustc-hex" && cd ..
+  - cd uint/ && cargo test --all-features && cargo test --no-default-features && cd ..
+  - cd keccak-hash/ && cargo test --no-default-features && cd ..
   - cd plain_hasher/ && cargo test --no-default-features && cd ..
   - cd parity-util-mem/ && cargo test --features=estimate-heapsize && cd ..
   - cd parity-util-mem/ && cargo test --features=jemalloc-global && cd ..
   - cd parity-util-mem/ && cargo test --features=mimalloc-global && cd ..
+  - cd rlp/ && cargo test --no-default-features && cargo test --benches && cd ..
```
Comment (ordian) [inline .travis.yml]:
```suggestion
  - cd rlp/ && cargo test --no-default-features && cargo check --benches && cd ..
```

**Comment by koushiro** [comment]
> Overall it looks good.
> 
> We have this ongoing (implicit) discussion about `rustc-hex`. It is a low-priority long-term goal to remove it from all Parity projects so the changes here that re-introduces it are a problem. Can you elaborate on why you seem to dislike `hex-literal` and prefer `rustc-hex`?

I just want to reduce the number of crates, if `rustc-hex` will be removed in the future, then I will revert it.

**Comment by koushiro** [comment]
@dvdplm  I want to know if you will release a new version for `rlp` crate on crates.io

**Comment by dvdplm** [comment]
@koushiro we will. Our policy is to test changes in `parity-ethereum` before releasing new versions. Once we have a PR there that passes all tests I'm fine making a release. Works for you?

**Comment by koushiro** [comment]
@dvdplm Thanks for your response, it works for me :)

**Comment by niklasad1** [comment]
@koushiro can you please fix the conflicts? Then we can merge it nice work  :+1:

**Comment by koushiro** [comment]
@niklasad1 Done :)

**Pull request merged**